src/errors.rs

```diff
@@ -39,7 +42,26 @@ pub(crate) trait PythonException<T> {
         let caller = Location::caller();
         self.into_pyresult().map_err(|err| {
             Python::with_gil(|py| {
-                PyErr::from_type(err.get_type(py), build_message(py, caller, &err, msg))
+                // Python treats KeyError differently then others:
+                // the message is always quoted, so that in case the key
+                // is an empty string, you see:
+                //
+                //   KeyError: ''
+                //
+                // instead of:
+                //
+                //   KeyError:
+                //
+                // This means that our message will be quoted if we reraise
+                // it as it is. So in this case we raise a RuntimeError instead.
+                if err
+                    .get_type(py)
+                    .is(pyo3::types::PyType::new::<pyo3::exceptions::PyKeyError>(py))
+                {
+                    PyRuntimeError::new_err(build_message(py, caller, &err, msg))
```

Comment:
This is another place where I do something "surprising", and I'm not sure if we want this.
Python special cases KeyError to quote the error message, so our nicely formatted error becomes a string with `\n` ecc.
To avoid this, I change the error to a PyRuntimeError, which "fixes" the problem with the error representation, but makes it impossible to catch any KeyError on the Python side. Which in general should be ok because if the dataflow run fails, we don't really want users to catch the exception and try again in the same process, but it still is a surpriing behaviour.